huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_workload_plan_stage_test.go

```diff
@@ -0,0 +1,181 @@
+package dws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getWorkloadPlanStageResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	var (
+		region  = acceptance.HW_REGION_NAME
+		httpUrl = "v2/{project_id}/clusters/{cluster_id}/workload/plans/{plan_id}"
+		product = "dws"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DWS client: %s", err)
+	}
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{cluster_id}", state.Primary.Attributes["cluster_id"])
+	getPath = strings.ReplaceAll(getPath, "{plan_id}", state.Primary.Attributes["plan_id"])
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	resp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		if _, ok := err.(golangsdk.ErrDefault403); ok {
+			return nil, golangsdk.ErrDefault404{}
+		}
+		return nil, fmt.Errorf("error retrieving DWS workload plan stage: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	respErrCode := utils.PathSearch("error_code", respBody, "")
+	if respErrCode != "" {
+		return nil, fmt.Errorf("error retrieving DWS workload plan stage: %s", utils.PathSearch("workload_res_str", respBody, nil))
+	}
```

Comment:
we cannot guarantee that all error codes have a corresponding `workload_res_str`. please make sure that nil is not returned in this case.